runtime/compiler/net/CommunicationStream.cpp

```diff
@@ -73,6 +73,13 @@ CommunicationStream::readMessage(Message &msg)
 
    // rebuild the message
    msg.deserialize();
+
+   // collect message size
+#ifdef STATS
+   static TR_Stats statMsgSize[MessageType_ARRAYSIZE];
+   statMsgSize[int(msg.type())].update(messageSize);
+   statMsgSize[int(msg.type())].report(stderr);
```

Comment:
This implementation prints something for every message that is received.
What we want is to collect these stats and print them one, it shutdown, much like we print stats about the number of messages by type